openml/testing.py

```diff
@@ -104,9 +112,39 @@ def tearDown(self):
         openml.config.server = self.production_server
         openml.config.connection_n_retries = self.connection_n_retries
 
+    @classmethod
+    def _track_test_server_dumps(self, entity_type, entity_id):
+        """ Static record of entities uploaded to test server
+
+        Dictionary of lists where the keys are 'entity_type'.
+        Each such dictionary is a list of integer IDs.
+        For entity_type='flow', each list element is a tuple
+        of the form (Flow ID, Flow Name).
+        """
+        if entity_type not in TestBase.tracker:
+            TestBase.tracker[entity_type] = [entity_id]
+        else:
+            TestBase.tracker[entity_type].append(entity_id)
+
+    @classmethod
+    def _delete_entity_from_tracker(self, entity_type, entity):
+        if entity_type in TestBase.tracker:
+            # delete_index handles duplicate entries
+            delete_index = []
+            for i, element in enumerate(TestBase.tracker[entity_type]):
+                if entity_type == 'flow':
+                    id, name = element
+                else:
+                    id = element
+                if id == entity:
+                    delete_index.append(i)
+            TestBase.tracker[entity_type] = [TestBase.tracker[entity_type][index]
```

Comment:
This appears to be overly complex. Can't you simply do list.remove(i)?

Reply:
I agree completely. It became quite unnecessarily complex. Since for flows I am maintaining records as list of (ID, name) whereas for other entity types it is just a list of IDs. 
Tried to keep the code as lean as possible.
I'll try another design and iterate once more.

Reply:
Have simplified this section a bit.
However, couldn't quite work out how to avoid the if-condition without changing how the file IDs are being tracked or recorded. That is, making the list of integers a list of a single-element tuple. Which I felt was more counter-intuitive and difficult to maintain/debug.